File: php/examples/history-example.php
<?php

    ## Capture Publish and Subscribe Keys from Command Line
    $publish_key   = isset($argv[1]) ? $argv[1] : false;
    $subscribe_key = isset($argv[2]) ? $argv[2] : false;

    # Print usage if missing info.
    if (!($publish_key && $subscribe_key)) {
echo("
    ==============
    EXAMPLE USAGE:
    ==============
    php ./publish-example.php PUBLISH-KEY SUBSCRIBE-KEY

");
        exit();
    }

    ## Require Pubnub API
    echo("Loading Pubnub.php Class\n");
    require('../Pubnub.php');

    ## -----------------------------------------
    ## Create Pubnub Client API (INITIALIZATION)
    ## -----------------------------------------
    echo("Creating new Pubnub Client API\n");
    $pubnub = new Pubnub( $publish_key, $subscribe_key );

    ## Get History
    echo("Requesting History...\n");
    $messages = $pubnub->history(array(
        'channel' => 'hello_world', ## REQUIRED Channel to Send
        'limit'   => 100            ## OPTIONAL Limit Number of Messages
    ));
    var_dump($messages);            ## Prints Published Messages.

?>

File: ruby/examples/history-example.rb
## -----------------------------------
## PubNub Ruby API History Example
## -----------------------------------

## including required libraries
require 'rubygems'
require './lib/pubnub.rb'

## declaring publish_key, subscribe_key, secret_key, cipher_key and ssl flag
publish_key   = 'demo'
subscribe_key = 'demo'
secret_key    = 'demo'
cipher_key    = 'demo'
ssl_on = true

## Print usage if missing info.
if !subscribe_key
  puts('
  Get API Keys at http://www.pubnub.com/account
  ==============
  EXAMPLE USAGE:
  ==============
  ruby history-example.rb PUBLISH-KEY SUBSCRIBE-KEY SSL-ON
  ruby history-example.rb demo demo true
  ')
end

## Create Pubnub Client API (INITIALIZATION)
puts('Initialize new Pubnub state')
pubnub = Pubnub.new(publish_key,subscribe_key,secret_key,cipher_key,ssl_on)

## Request Past Publishes (HISTORY)
puts('Requesting History with history() Function')
message = pubnub.history(
{
  'channel' => 'hello_world',
  'limit'   => 3
})
puts(message)
